### ISSUE TITLE
update process repeatedly adds itself to Windows system startup tasks

### ISSUE THREAD

**Issue by mjsulik** [lmstudio-ai/lmstudio-bug-tracker#209]
lmstudio should have an option to disable automatically running startup processes and should respect the user's choice to disable automatic startup on Windows (e.g., using the Windows task manager). The current behavior of re-adding disabled startup tasks when the app is opened or updated (I'm unclear about what exactly triggers this behavior) is an approach commonly used by malware and raises red flags.

**Comment by nuclearsugar** [comment]
Seconded. Please add a checkbox in the settings window that will disable the automatically running startup processes.

**Comment by BihartGames** [comment]
I completely agree on this need of user determination of startup tasks rather than the developers choice. please add the feature for users to decide whether they want LM Studio to startup on system boot, or not. As a user of LM Studio myself, I hate when apps startup on boot as it slows the system down more than some might think.

**Comment by yagil** [comment]
There’s a checkbox on settings called “local LLM service” which controls this. If it resets after an update it’s (obviously) a bug. Did you uncheck it and it still happens?

See here for where to find it: https://lmstudio.ai/docs/advanced/headless#run-the-llm-service-on-machine-login

**Comment by mjsulik** [comment]
Thank you for clarifying! I think the confusion was caused for myself and others by the the label for this checkbox. I  suggest changing it to something like "Start automatically."

**Comment by yagil** [comment]
> Thank you for clarifying! I think the confusion was caused for myself and others by the the label for this checkbox. I suggest changing it to something like "Start automatically."
> 
> Edit: After looking at the documentation you linked, I don't think your answer addressed my suggestion. The documentation says: "When this setting is enabled, exiting the app will minimize it to the system tray, and the LLM server will continue to run in the background." I'm suggesting that there should be an option for both the LLM server and the GUI to NOT run automatically.

Hi @mjsulik see highlighted text here

<img width="810" alt="image" src="https://github.com/user-attachments/assets/eecd2b42-21a9-44dd-8a2d-6f725645fa09">


I'd like the understand if this is a wording confusion or an actual bug, because "re-adding disabled startup tasks" is not expected. Is this still happening?

**Comment by hiccups1980** [comment]
To be honest, this feels like a viral behavior. I've been using Windows since Windows 3.11 and I don't know of a single program that can change the decision whether it should autostart or not (set via task manager) itself.
I deactivate the autostart via the task manager and the program reactivates its autostart on its own. This is a very horrifying bug.

**Comment by Xecular** [comment]
It's definitely changing startup settings without user interaction. I disabled this behavior when I installed lm studio several weeks ago, but it started doing it again as of my last restart.

It would be better to make use of the task manager to control this functionality intended instead of with an in-app setting

**Comment by hiccups1980** [comment]
> It would be better to make use of the task manager to control this functionality, since the user is intended to be able to control all startup settings through it

As I said. The program is able to manipulate the autostart task manager setting. If you set it to "deactivate" in the task manager it will "activate" the autostart by itself. 
I don't know how to describe it any more precisely. The program has the ability to monitor the autostart setting in the task manager and will immediately reactivate the autostart if you try to stop it from autostarting via the task manager.
As described, for me this is part of a virus behavior and therefore NOT ACCEPTABLE!

**Comment by yagil** [comment]
1. It's intended to be the user's choice whether to have the service to run at start up.
2. **If** somehow it gets re-added after you remove it (haven't verified that yet), that would be due to a bug, obviously. I say obviously because it should be obvious to you it's not something we'd want to do.
3. This github issue with 5 participants and 3 likes is the only report I've seen about this. I'm going to test this now and verify whether it happens, and will update you with the results

**Comment by yagil** [comment]
I just tested this and this is **not** reproducing for me on Windows. I used [LM Studio 0.3.5](https://lmstudio.ai/download), which has been on the website for the last couple months. 


You can perform the test to verify yourself as well:

1. head to app settings and deselect "run as service" **IF** it is selected. 
2. restart your machine
3. upon start up, verify LM Studio is indeed not running

If you can demonstrate the test above is not working the same way for you, we will investigate more.

<img width="868" alt="image" src="https://github.com/user-attachments/assets/7ec9bf36-42e8-47ef-8d01-dc1c62eb57d3" />

For the record the app doesn't monitor or manipulate anything to achieve this, it's a standard Electron (the framework on which the app is built) API call.

**Comment by yagil** [comment]
Moreover upon reviewing our code, we specifically won't override the user's preference. So even if you keep that checkbox checked inside the app + disable it via Task Manager yourself, the app will not "re-add" itself.  If you can demonstrate what I'm saying here isn't correct, please comment back and we will investigate. Until then I'm closing this issue.

**Comment by hiccups1980** [comment]
You can easily reproduce it:

- Install LM Studio
- Do not change anything in the settings directly in LM Studio (leave the service activated as it is the default setting)
- Exit the LM Studio
- Deactivate the “update.exe” in “Autostart” in the task manager
- Shut down your PC (do not restart!)
- Start up your PC again

That was it! LM Studio will start automatically and the “update.exe” from LM Studio will be listed as “ enabled” in the task manager under “autostart”.
This is a strange and very aggressive behavior.

**Comment by AMISS-DEV** [comment]
> You can easily reproduce it:
> 
> * Install LM Studio
> * Do not change anything in the settings directly in LM Studio (leave the service activated as it is the default setting)
> * Exit the LM Studio
> * Deactivate the “update.exe” in “Autostart” in the task manager
> * Shut down your PC (do not restart!)
> * Start up your PC again
> 
> That was it! LM Studio will start automatically and the “update.exe” from LM Studio will be listed as “ enabled” in the task manager under “autostart”. This is a strange and very aggressive behavior.
> 
> I just did these steps and I could reproduce it again.

This is happening with my install as well, and is 100% reproducible.

**Comment by yagil** [comment]
Can you please share screenshots of the relevant part in Task Manager?

**Comment by AMISS-DEV** [comment]
![{D597AD37-C6BC-4C62-B232-0D003DB6291B}](https://github.com/user-attachments/assets/232602f7-beb5-48c7-9fbb-873a8401d07a)


I set the Update.exe Startup app to "Disabled". However, when I start my PC back up it is set to "Enabled" once again and LM Studio is running in the background. I've done this probably ~10 times now and it always re-enables itself.

**Comment by yagil** [comment]
Thanks for this. It’s interesting it says Update.exe. Incidentally we just moved to a new installer that doesn’t rely on Update.exe (from a framework called “Squirrel”, fyi), so maybe — just speculating since I’m not seeing this — whatever corner case this might be got solved?

Would you be able to give it a try?  Available here: 

https://installers.lmstudio.ai/win32/x64/0.3.5-10/LM-Studio-0.3.5-10-x64.exe

**Comment by AMISS-DEV** [comment]
That seems to have done the trick! Update.exe has been replaced by LM Studio.exe in the Startup apps section of the Task Manager, and it is retaining its "Disabled" behavior on restart.

I appreciate your prompt reply, and keep up the good work. Love what you all are doing.

**Comment by yagil** [comment]
@AMISS-DEV thank you! This is great news. Thanks for testing it out. Happy holidays! 🎄

We’ll be publishing our new installer with v0.3.6. If anyone is hitting this in the meantime, see beta build link above.

**Comment by hiccups1980** [comment]
I appreciate your work and the update. 👍

**Comment by zshmoney** [comment]
I am still having is same issue with windows 10 pro. It says LM Studio not update but is continuing to change my startup preference for LM Studio. This is the only app I have ever seen reactivate it, and multiple times. Other than that I love the program but this is a concerning issue/bug.

**Comment by zshmoney** [comment]
Yes that is the version I am currently using. It has reactivated itself twice on Windows 10 Pro

**Comment by yagil** [comment]
Anyone else who previously ran into this seeing what @zshmoney is reporting?

@zshmoney it would help if you could attach a video of the reproduction

**Comment by alyandon** [comment]
> Anyone else who previously ran into this seeing what [@zshmoney](https://github.com/zshmoney) is reporting?
> 

100% reproducible here on Windows 10.  The 0.3.8 version of LM Studio re-adds the HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\"electron.app.LM Studio" entry every time I execute the program.  There is similar behavior regarding the updater process as well.

Steps to reproduce:

1) Install LM Studio
2) Exit LM Studio
3) Disable LM Studio as a startup app
4) Open regedit.exe and navigate to HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion
5) Verify there is not an "electron.app.LM Studio" key present
6) Run LM Studio again
7) See that an auto-run key for "electron.app.LM Studio" has been created in HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion

**Comment by SublimePeace** [comment]
Can confirm  what @alyandon is saying, I reproduced this using Autoruns (Sysinternals). No option to disable auto-run on startup in the settings, and the app creates an auto-run registry key every time it launches. This is straight up malware behavior.

I've never in my life seen a software default to auto run on startup, while at the same time not having a setting to turn that off. Let alone have it make an auto-start registry key every time it's launched.

**Comment by yagil** [comment]
And are you sure toggling this off in settings doesn’t turn this off?  

![Image](https://github.com/user-attachments/assets/601529be-151f-4922-8181-16fd743b7649)

@alyandon @SublimePeace 

To be clear we’re not using anything special other than Electron standard API for this: https://www.electronjs.org/docs/latest/api/app#appgetloginitemsettingsoptions-macos-windows

Software like Slack, Spotify, Figma etc will likely do the same when configured to launch on machine startup.

**Comment by alyandon** [comment]
@yagil 

Yes - it appears toggling the "Local LLM Service" option off seems to stop LM Studio from creating an autostart entry in the registry every time it is run.

Regardless, "wanting LM Studio to minimize to the system tray and continue running in the background" is the behavior I actually want while I'm running my local tooling that needs the API.  That is very very different from "auto-registering as a startup app in Windows every time the user starts LM Studio even if the user has previously disabled it as a startup application".

**Comment by AaronLil** [comment]
I was able to fix this issue by leaving `Enable Local LLM Service` disabled and adding the `--run-as-service` parameter to the Start menu (or Desktop) shortcut. Follow the link: https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/303#issuecomment-2641366239

**Comment by rigopoui** [comment]
This just happened to me after an update. Uninstalling and will not use software that doesn't respect the user's choices.

**Comment by yagil** [comment]
> This just happened to me after an update. Uninstalling and will not use software that doesn't respect the user's choices.

Kindly mention which version you updated from and to, and what you saw that makes you state it didn’t respect your choice

**Comment by SRaresP** [comment]
If the original issue which dictates the open/closed state of the issue is that the app turns it's startup back on with the "Enable local llm service" ON, it's happening again with the same steps as the last mentioned above, and the issue should be reopened.

**Comment by Qbsoon** [comment]
Keeps happening to me also on 0.3.18 and w11 (10.0.22631).
If I disable "Local LLM Service", it stops, but I want it to run as a service, but only when I open it, not to be run with every startup.
If I enable "Local LLM Service" and disable startup in task manager, after reboot LM Studio still opens and it's startup is enabled in the task manager. As others said, this is very shady behavior I also never saw with any other app.

**Comment by SRaresP** [comment]
> Keeps happening to me also on 0.3.18 and w11 (10.0.22631). If I disable "Local LLM Service", it stops, but I want it to run as a service, but only when I open it, not to be run with every startup. If I enable "Local LLM Service" and disable startup in task manager, after reboot LM Studio still opens and it's startup is enabled in the task manager. As others said, this is very shady behavior I also never saw with any other app.

Yea it appears that they simply don't care, since more than a month passed since my reply. If you create a new issue they will probably just mark it as a dupe and call it a day...

I mean damn, I appreciate the project since it's free, but if they cared about the community they would at least keep this issue open until they do something about it.

**Comment by Qbsoon** [comment]
> > Keeps happening to me also on 0.3.18 and w11 (10.0.22631). If I disable "Local LLM Service", it stops, but I want it to run as a service, but only when I open it, not to be run with every startup. If I enable "Local LLM Service" and disable startup in task manager, after reboot LM Studio still opens and it's startup is enabled in the task manager. As others said, this is very shady behavior I also never saw with any other app.
> 
> Yea it appears that they simply don't care, since more than a month passed since my reply. If you create a new issue they will probably just mark it as a dupe and call it a day...
> 
> I mean damn, I appreciate the project since it's free, but if they cared about the community they would at least keep this issue open until they do something about it.

And it's really strange for an open project to keep such a behavior. I get it might be not intended somehow, but it's strange to not remove it when pointed out.

**Comment by yagil** [comment]
This is a bug in 0.3.18 caused by updating the underlying Electron version. Fixed in 0.3.19, available in beta and being released in stable today

Beta available here: https://lmstudio.ai/beta-releases

**Comment by yagil** [comment]
0.3.19 is now out with the fix: https://lmstudio.ai/blog/lmstudio-v0.3.19

**Comment by Qbsoon** [comment]
Welp, I just updated, turned on "Local LLM Service" to check it and it still doesn't work as it should. It's now very strange.
1. I've turned on the service
2. LM Studio turned itself on in task manager autostart list
3. I disabled it on the list
4. The "Local LLM Service" option was automatically disabled
5. I turn on "Local LLM Service" option
6. Etc. in a loop

So to be honest it's the same as before in practice, because service still turns on autostart and still can't have both autostart off and service option on.

**Comment by yagil** [comment]
> Welp, I just updated, turned on "Local LLM Service" to check it and it still doesn't work as it should. It's now very strange.
> 
> 1. I've turned on the service
> 2. LM Studio turned itself on in task manager autostart list
> 3. I disabled it on the list
> 4. The "Local LLM Service" option was automatically disabled
> 5. I turn on "Local LLM Service" option
> 6. Etc. in a loop
> 
> So to be honest it's the same as before in practice, because service still turns on autostart and still can't have both autostart off and service option on.

Hi, are you able to share a video showing this? And to make sure, you’re on 0.3.19 build 4?

**Comment by Qbsoon** [comment]
> 
> Hi, are you able to share a video showing this? And to make sure, you’re on 0.3.19 build 4?



Ok. I didn't know where to post it, so I uploaded it to youtube as a link-only video.
I showcased three loops. Second one is the one I described above, but on the first loop I found something more by accident and it is also shown on loop third.
I forgot to mention in the steps that I've been shutting down LM Studio before opening the task manager. Well in the 1st and 3rd loop I didn't and it turns out that then it is both autostart disabled and service enabled, but only for this session. As soon as I restart the LM Studio, service option is disabled. And when I re-enable it, again autostart enabled and the loop persists.
[https://www.youtube.com/watch?v=ZbfuthA7JhE](https://www.youtube.com/watch?v=ZbfuthA7JhE)

Edit: Forgot to mention, but it's visible on the video that the version is correct. (In the window title)
